packages/cf.js/test/fixture.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  AppInstanceInfo,
+  AssetType,
+  INodeProvider,
+  Node
+} from "@counterfactual/common-types";
+import { BigNumber } from "ethers/utils";
+import EventEmitter from "eventemitter3";
+
+export class TestNodeProvider implements INodeProvider {
+  public postedMessages: Node.Message[] = [];
+  readonly callbacks: ((message: Node.Message) => void)[] = [];
+  readonly messageEmitter: EventEmitter = new EventEmitter();
```

Comment:
Is there really enough commonality with the real `NodeProvider` for both TestNodeProvider and NodeProvider to inherit from a superclass?

Reply:
I would rather prefer redundancy over dubious inheritance. Seems a bit forced.

Reply:
@mvanderh well I'm assuming that the 3 lines above are literally line for line copied from an instantiation of a real `NodeProvider and thus if we were to change any character in those 3 lines it would break and not be a good test case?

Reply:
The purpose of a test is to break when we're changing stuff like that, I believe. It makes us be conscious of API changes. Test concerns aren't supposed to affect class architecture.

Reply:
@snario They are not copied from the real NodeProvider. TestNodeProvider and real NodeProvider are structered very differently AFAICT.